tests/test-verify-volume-device.yml

```diff
@@ -1,4 +1,25 @@
 ---
+
+#
+# MDRaid devices paths are returned as a symlinks. But sometimes we need their targets.
+#
+- name: (1/3) Process device path (set initial value)
+  set_fact:
+    storage_test_device_path: "{{ storage_test_volume._device }}"
+
+# realpath fails when given empty string; the task then returns completely different dict.
+- block:
+  - name: (2/3) Process device path (get device file info)
+    command: realpath "{{ storage_test_volume._device }}"
+    register: storage_test_realpath
+
+  - name: (3/3) Process device path (replace device with its target if it is a symlink)
+    set_fact:
+      storage_test_device_path: "{{ storage_test_realpath.stdout }}"
+    when: storage_test_device_path not in storage_test_blkinfo.info
+
+  when: storage_test_volume._device != ""
```

Comment:
I have encountered a situation when it actually happened. `realpath` then returns non-zero exit code and ansible registers dict with different set of values. It will eventually result in completely unrelated error message which is hard to understand.